unitTests/ui/MissingTermManagerTest.ts

```diff
@@ -0,0 +1,65 @@
+import * as Mock from '../MockEnvironment';
+import { MissingTermManager } from '../../src/ui/MissingTerm/MissingTermManager';
+import { Simulate } from '../Simulate';
+import { $$, l } from '../Test';
+
+export function MissingTermsManagerTest() {
+  describe('MissingTermManager', () => {
+    let env: Mock.IMockEnvironment;
+    let arrayGet: Array<string>;
+    const spyGet = () => {
+      return arrayGet;
+    };
+    const spySet = (key: any, values: any) => {
+      arrayGet = values;
+    };
+    const populateBreadcrumb = () => {
+      const missingTerm = 'is';
+      arrayGet.push(missingTerm);
+      return Simulate.breadcrumb(env);
+    };
+
+    beforeEach(() => {
+      env = new Mock.MockEnvironmentBuilder().build();
+      new MissingTermManager(env.root, env.queryStateModel, env.queryController);
+      env.queryStateModel.get = jasmine.createSpy('missingTermSpy').and.callFake(spyGet);
+      env.queryStateModel.set = jasmine.createSpy('missingTermSpy').and.callFake(spySet);
+      arrayGet = [];
+    });
+
+    it('add missing term from the url in advance query', () => {
+      const missingTerm = 'is';
+      arrayGet.push(missingTerm);
+      const query = Simulate.query(env, {
```

Comment:
I'd prefer to call this const `simulation` since it contains more stuff than just the "query"